src/thor/astar.cc

```diff
@@ -432,6 +432,7 @@ void AStarPathAlgorithm::SetOrigin(GraphReader& graphreader,
             cost.cost += dest_path_edge.distance();
             cost.cost = std::max(0.0f, cost.cost);
             dist = 0.0;
+            break;
```

Comment:
this loop expected (correctly) that it would never see any duplicate edge candidates so it didnt need to break once it found the matching one, but its was wasteful to keep looping over ids that it didnt need to. however because of the bug in filtering pbf marshalling, the same edge is seen twice when we relax the astarheuristic and run the route a second time. this means that that subtraction above `cost -= remainder_cost;` happens twice which makes the cost negative :womp-womp: